src/compare-oracle-to-fallback/compare-oracle-to-fallback.spec.js

```diff
@@ -0,0 +1,167 @@
+// required libraries
+const BigNumber = require('bignumber.js');
+const { createBlockEvent } = require('forta-agent');
+const { provideHandleBlock, createAlert } = require('./compare-oracle-to-fallback');
+
+const SECONDS_PER_DAY = 86400;
+
+describe('AAVE oracle versus fallback oracle agent', () => {
+  let handleBlock;
+
+  // this function allows us to mock the behavior of the initializeTokensContracts function
+  // in production, that function:
+  function mockTokensContractsAlertsPromise(assetPriceOracle, assetPriceFallback, tStart) {
+    // create a token
+    const reserveToken = { symbol: 'FAKE1', tokenAddress: '0xFIRSTFAKETOKENADDRESS' };
+
+    // create a mock price oracle contract for returning the value we want to test with
+    const priceOracleContractInstance = {
+      getAssetPrice: jest.fn(() => Promise.resolve(assetPriceOracle)),
+    };
+
+    // create a mock fallback oracle contract for returning the value we want to test with
+    const fallbackOracleContractInstance = {
+      getAssetPrice: jest.fn(() => Promise.resolve(assetPriceFallback)),
+    };
+
+    // create an object to track the number of token alerts in the last day
+    const tokenAlert = { numAlertsInLastDay: 0, tStart };
+
+    // create the Array of Tuples of tokens / oracle / fallback / alerts objects
+    const tokenContractFallbackAlertTuples = [
+      {
+        reserveToken, priceOracleContractInstance, fallbackOracleContractInstance, tokenAlert,
+      },
+    ];
+
+    // wrap the Array of Tuples in a Promise (which we will resolve immediately)
+    // the original function returns a Promise that is resolved when all contract interactions have
+    // finished
+    return Promise.resolve(tokenContractFallbackAlertTuples);
+  }
+
+  function calculatePercentError(first, second) {
```

Comment:
Consider making this a shared function since you do the identical operation inline in the agent code

Reply:
This function and two others have been moved to `agent-setup.js`.